documentation/specs/Toast.md

```diff
@@ -0,0 +1,321 @@
+# `Toast` Component Specification
+
+## Overview
+
+The `Toast` component displays a brief non-disruptive message to the user as a result of an action taken. The displayed message is accompanied by an associated status icon and cannot be manually dismissed. **Not** to be used to display critical information such as a system failure, see the `Alert` component for such behavior.
+
+### Use Cases
+
+- Inform user of a completed action such as an account creation.
+- Inform user that an asynchronous process has been triggered based on their action.
+
+### Features
+
+- Individual toasts have ARIA `role="status"`.
+- The `message` prop is of type `string`.
+
+### Risks and Challenges
+
+- Unlike other components, `Toast` components are unique in that they may be quite opinionated with regards to how they display and the kind of control they expose to consumers.
+- Building for the shipper's experience at EasyPost according to internal specs while maintaining a pleasant DX for outside consumers.
+
+### Prior Art
+
+- [Paste `<Toast />`](https://paste.twilio.design/components/toast)
+- [Spectrum `<Toast />`](https://react-spectrum.adobe.com/react-spectrum/Toast.html)
+
+---
+
+## Design
+
+### API
+
+```typescript
+import type { AriaToastProps } from "@react-aria/toast";
+import type { QueuedToast } from "@react-stately/toast";
+
+export type ToastStatus =
+  | "primary"
+  | "success"
+  | "secondary"
+  | "error"
+  | "warning";
+
+export type ToastProps = {
+  /**
+   * Toast status
+   * @default "success"
+   */
+  status?: ToastStatus;
+  /**
+   * Toast will render with left aligned status icon
+   * @default false
+   */
+  noIcon?: boolean;
+  /**
+   * Toast message
+   */
+  message: string;
+};
+
+export type ToastStateProps = AriaToastProps<ToastProps> & {
+  toast: QueuedToast<ToastProps>;
+  state: ToastState<ToastProps>;
+};
+```
+
+### Anatomy
```

Comment:
anatomy is missing some details like name destructuring from props, passing props to `Text`, and passing `symbol` to `Icon`. let me know if you'd like to see them added now as opposed to returning to add them later